java/arcs/tools/Recipe2Plan.kt

```diff
@@ -0,0 +1,57 @@
+package arcs.tools
+
+import arcs.core.storage.api.DriverAndKeyConfigurator
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.arguments.argument
+import com.github.ajalt.clikt.parameters.arguments.multiple
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.flag
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.types.file
+import com.squareup.kotlinpoet.FileSpec
+import java.io.File
+
+/** Generates plans from recipes. */
+class Recipe2Plan : CliktCommand(
+    help = """Generate plans from recipes.
+    
+    This script reads serialized manifests and generates Kotlin files with [Plan] classes.""",
+    printHelpOnEmptyArgs = true
+) {
+    private val outdir by option(help = "output directory; defaults to '.'").file(fileOkay = false)
```

Comment:
I'd recommend `--outfile` instead. Bazel is incredibly weird about its folders and files, it's usually easier to just tell the binary exactly where to put things (even piping it from stdout is an ok approach) from your BUILD rule, rather than having the binary be smart about it.

The BUILD rule is the only place where you have the full context around what files should go where, so that's where you should do the `s/.bin.pb/.jvm.kt/` replacement.

Also the comment here says the default is `'.'`, but I can't see where that's set...?

Reply:
I structured the CLI to take multiple proto files in and return multiple generated Kotlin files out. To maintain this, specifying output files over directories would really complicate the build rules / CLI arguments (pass in multiple lists of files?) -- unless you have an alternative recommendation that I'm not seeing. 

One alternative would be to take in multiple proto files and return one generated plan. I'm looking over how that would look now, and...
I think that could work. This makes sense since `arcs_kt_plan_2` ends by feeding all the generated code into a single jar. I'll start working on this. 

(I'm trying to think "aloud" in comments, in the spirit of over-communicating.)

Reply:
I'm curious why you've opted for that approach rather than for the simpler one where it takes one input file and generates one output file?

Designing the BUILD rules to make effective use of a multiple-files-in-multiple-files-out approach seems complex. Bazel's approach to parallel builds means invoking the binary once per file should be done really efficiently, so I wouldn't worry about speed for this. In fact, for incremental builds, it's preferable to have one invocation per input file, so that untouched files don't get processed again.

Reply:
I've since pivoted towards a 1:1 approach. You make a persuasive argument. This may change in the future (as I get a better grip of depsets, providers, and aspects), but seems very reasonable for now.